sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -133,6 +134,15 @@ static void _on_sock_evt(sock_udp_t *sock, sock_async_flags_t type, void *arg)
     coap_pkt_t pdu;
     sock_udp_ep_t remote;
     gcoap_request_memo_t *memo = NULL;
+    /* Code paths that necessitate a response on the message layer can set a
+     * response type here (COAP_TYPE_RST or COAP_TYPE_ACK). If set, at the end
+     * of the function there will be
+     *   * that value will be put in the code field,
+     *   * token length cleared,
+     *   * code set to EMPTY, and
+     *   * the message is returned with the rest of its header intact.
+     */
+    int8_t messagelayer_emptyresponse_type = -1;
```

Comment:
Please use a macro for `-1` and further below in the code where the value is checked.